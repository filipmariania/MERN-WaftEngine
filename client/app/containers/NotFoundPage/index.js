/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import logo from '../../assets/img/logo-icon.svg';
import './404.css';

/* eslint-disable react/prefer-stateless-function */
export default class NotFound extends React.PureComponent {
  render() {
    return (
      <div className="py-10 mx-auto text-center overflow-hidden h-full">
        <h1 className="font-light text-4xl">
          <FormattedMessage {...messages.header} />
        </h1>
        <div className="logo-swing">
          <img src={logo} className="logo opacity-25" />
          <img id="hinge" src={logo} className="logo hinge" />
        </div>
      </div>
    );
  }
}
