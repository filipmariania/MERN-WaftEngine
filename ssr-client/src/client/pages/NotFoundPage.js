import React from 'react';

// eslint-disable-next-line react/prop-types
const NotFoundPage = ({ staticContext = {} }) => {
  staticContext.notFound = true; // eslint-disable-line no-param-reassign
  return <h1>Ooops, route not found.</h1>;
};

export default {
  component: NotFoundPage,
};
