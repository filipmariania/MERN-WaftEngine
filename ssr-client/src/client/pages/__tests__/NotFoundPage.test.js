import React from 'react';
import { create } from 'react-test-renderer';
import NotFoundPage from '../NotFoundPage';

test('snapshot', () => {
  const Component = NotFoundPage.component;
  const c = create(<Component />);
  expect(c.toJSON()).toMatchSnapshot();
});
