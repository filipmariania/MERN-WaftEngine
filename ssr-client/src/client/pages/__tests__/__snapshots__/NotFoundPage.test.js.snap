// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<h1>
  Ooops, route not found.
</h1>
`;
